### PR TITLE
NAS-110589 / 21.06 / UI does not show selected pool in advanced settings for applications

### PR DESCRIPTION
**Pull request by AlexKarpov98**

### CODE REVIEW

File: src/app/pages/applications/forms/kubernetes-settings.component.ts [hunk 1]
```diff
@@ -101,29 +101,53 @@ export class KubernetesSettingsComponent implements FormConfiguration {
...
+        resolve(true);
+      },
+      (err) => {
+        resolve(false);
       });
     });
+
+    return await Promise.all([promise1, promise2, promise3]).then(
+      (res) => true,
+    );
```
Comment (undsoft) [inline src/app/pages/applications/forms/kubernetes-settings.component.ts:146]: I know that the `entity-form.interface.ts` uses promises for prerequisite, but I think we should be using observables when possible. They are more natural to Angular ecosystem and we will have less work, if we decide to start using ngrx.

Also, please correct naming of variables when doing so to make them descriptive and match camelCase:

Example:
```typescript
const setPoolControl = this.appService.getPoolList().pipe(
  tap((pools) => {
    const poolControl = _.find(this.fieldSets[0].config, { name: 'pool' });
    pools.forEach((pool) => {
      poolControl.options.push({ label: pool.name, value: pool.name });
    });
  }),
);

const setNodeIpControl = this.appService.getBindIPChoices().pipe(
  tap((ips) => {
    const nodeIpControl = _.find(this.fieldSets[0].config, { name: 'node_ip' });
    for (const ip of ips) {
      nodeIpControl.options.push({ label: ip, value: ip });
    }
  }),
);

...

return forkJoin([setPoolControl, setNodeIpControl, ...])
  .pipe(
    map(() => true),
    // catchError may not be needed if you want to fail when something hasn't been loaded.
    catchError((error) => {
      console.log(error);
      return of(false);
    }),
  )
  .toPromise();
```

**Comment by AlexKarpov98** [comment]
@undsoft Fixed above issue.

File: src/app/pages/applications/forms/kubernetes-settings.component.ts [hunk 2]
```diff
@@ -101,29 +102,46 @@ export class KubernetesSettingsComponent implements FormConfiguration {
...
+        const v4_interface_control = _.find(this.fieldSets[1].config, { name: 'route_v4_interface' });
+        interfaces.forEach((i) => {
+          v4_interface_control.options.push({ label: i.name, value: i.name });
+        });
+      }),
+    );
+
+    return forkJoin([setPoolControl, setNodeIpControl, setV4InterfaceControl]).pipe(
+      map(() => true),
+      // catchError may not be needed if you want to fail when something hasn't been loaded.
```
Comment (undsoft) [inline src/app/pages/applications/forms/kubernetes-settings.component.ts:135]: This was a comment for you, not for final code :)

File: src/app/pages/applications/forms/kubernetes-settings.component.ts [hunk 3]
```diff
@@ -101,29 +102,46 @@ export class KubernetesSettingsComponent implements FormConfiguration {
...
+        const node_ip_control = _.find(this.fieldSets[0].config, { name: 'node_ip' });
+        for (const ip in ips) {
+          node_ip_control.options.push({ label: ip, value: ip });
+        }
+      }),
+    );
+
+    const setV4InterfaceControl = this.appService.getInterfaces().pipe(
+      tap((interfaces: any[]) => {
+        const v4_interface_control = _.find(this.fieldSets[1].config, { name: 'route_v4_interface' });
```
Comment (undsoft) [inline src/app/pages/applications/forms/kubernetes-settings.component.ts:126]: Please update variables names to use camelCase

**Comment by AlexKarpov98** [comment]
@undsoft Fixed.

**Review by undsoft — APPROVED** [review]
It works, thanks.

Снимок экрана 2021-05-17 в 14 19 29

**Pull request merged**